packages/@aws-cdk/aws-elasticloadbalancingv2-actions/lib/cognito-action.ts

```diff
@@ -0,0 +1,89 @@
+import * as cognito from '@aws-cdk/aws-cognito';
+import * as elbv2 from '@aws-cdk/aws-elasticloadbalancingv2';
+import { Duration } from '@aws-cdk/core';
+
+/**
+ * Properties for AuthenticateCognitoAction
+ */
+export interface AuthenticateCognitoActionProps {
+  /**
+   * What action to execute next
+   */
+  readonly next: elbv2.ListenerAction;
+
+  /**
+   * The Amazon Cognito user pool.
+   */
+  readonly userPool: cognito.IUserPool;
+
+  /**
+   * The Amazon Cognito user pool client.
+   */
+  readonly userPoolClient: cognito.IUserPoolClient;
+
+  /**
+   * The domain prefix or fully-qualified domain name of the Amazon Cognito user pool.
+   */
+  readonly userPoolDomain: cognito.IUserPoolDomain;
+
+  /**
+   * The query parameters (up to 10) to include in the redirect request to the authorization endpoint.
+   *
+   * @default - No extra parameters
+   */
+  readonly authenticationRequestExtraParams?: Record<string, string>;
```

Comment:
I wasn't aware that this goes through JSII transformation correctly, does it? 

Why this over index type - `{ [key: string]: string } `?

Reply:
Because I hate typing that. This is equivalent.